File: AppConHilosJava/Programa1.java
public class Programa1 extends Thread{
    @Override
    public void run(){
        System.out.println(Thread.currentThread().getName() + " subproceso corriendo...");
    }

    public static void main(String[] args){
        Thread hilo = new Thread(new Programa1(), "Hilo1");
        hilo.start();
        System.out.println("Este codigo esta fuera del hilo");
    }
}


/*

@Override
    public void run(){
        System.out.println(Thread.currentThread().getName() + " subproceso corriendo...");
    }

    public static void main(String[] args){
        Programa1 hilo = new Programa1();
        hilo.start();
        System.out.println("Este codigo esta fuera del hilo");
    }
*/

File: AppConHilosPython/programa1.py
import threading
import os

def task1():
        print ("Tarea 1 asignada al thead: {}".format(threading.current_thread().name))
        print ("ID del hilo corriendo la tarea 1: {}".format(threading.current_thread().ident))

def task2():
        print (" Tarea 2 asignada al thead: {}".format(threading.current_thread().name))
        print ("ID del hilo corriendo la tarea 2: {}".format(threading.current_thread().ident))

if __name__ == "__main__":
        print ("ID del proceso corriendo el programa main: {}".format(os.getpid()))
        print ("ID del identificador del hilo corriendo el programa main: {}".format(threading.current_thread().ident))
        print ("Nombre del Main thread: {}".format(threading.current_thread().name))

        t1 = threading.Thread(target=task1, name="Hilo 1")
        t2 = threading.Thread(target=task2, name="Hilo 2")

        t1.start()
        t2.start()

        t1.join()
        t2.join()

        print("Programa terminado")


